qa/tasks/mgr/dashboard/test_cephfs.py

```diff
@@ -71,3 +71,68 @@ def test_cephfs_list(self):
         self.assertIn('mdsmap', cephfs)
         self.assertIsNotNone(cephfs['id'])
         self.assertIsNotNone(cephfs['mdsmap'])
+
+    def test_ls_mk_rm_dir(self):
+        data = self._get("/api/cephfs/1/ls_dir", params={'path': '/'})
+        self.assertStatus(200)
+        self.assertIsInstance(data, list)
+        self.assertEqual(len(data), 0)
+
+        self._post("/api/cephfs/1/mk_dirs", params={'path': '/pictures/birds'})
+        self.assertStatus(200)
+
+        data = self._get("/api/cephfs/1/ls_dir", params={'path': '/pictures'})
+        self.assertStatus(200)
+        self.assertIsInstance(data, list)
+        self.assertEqual(len(data), 1)
+
+        self._post("/api/cephfs/1/rm_dir", params={'path': '/pictures'})
+        self.assertStatus(500)
+        self._post("/api/cephfs/1/rm_dir", params={'path': '/pictures/birds'})
+        self.assertStatus(200)
+        self._post("/api/cephfs/1/rm_dir", params={'path': '/pictures'})
+        self.assertStatus(200)
+
+        data = self._get("/api/cephfs/1/ls_dir", params={'path': '/'})
+        self.assertStatus(200)
+        self.assertIsInstance(data, list)
+        self.assertEqual(len(data), 0)
+
+    def test_snapshots(self):
+        self._post("/api/cephfs/1/mk_dirs", params={'path': '/movies/dune'})
+        self.assertStatus(200)
+
+        self._post("/api/cephfs/1/mk_snapshot",
+                   params={'path': '/movies/dune', 'name': 'test'})
+        self.assertStatus(200)
+
+        data = self._get("/api/cephfs/1/ls_dir", params={'path': '/movies'})
+        self.assertStatus(200)
+        self.assertIsInstance(data, list)
+        self.assertEqual(len(data), 1)
+        self.assertSchema(data[0], JObj(sub_elems={
+            'name': JLeaf(str),
+            'path': JLeaf(str),
+            'snapshots': JList(JObj(sub_elems={
+                'name': JLeaf(str),
+                'path': JLeaf(str),
+                'created': JLeaf(str)
+            })),
+            'quotas': JObj(sub_elems={
+                'max_bytes': JLeaf(int),
+                'max_files': JLeaf(int)
+            })
+        }))
+        snapshots = data[0]['snapshots']
+        self.assertEqual(len(snapshots), 1)
+        snapshot = snapshots[0]
+        self.assertEqual(snapshot['name'], "test")
+        self.assertEqual(snapshot['path'], "/movies/dune/.snap/test")
+
+        self._post("/api/cephfs/1/rm_snapshot",
+                   params={'path': '/movies/dune', 'name': 'test'})
+        self.assertStatus(200)
+
+        data = self._get("/api/cephfs/1/ls_dir", params={'path': '/movies'})
+        self.assertStatus(200)
+        self.assertEqual(len(data[0]['snapshots']), 0)
```

Comment:
I'd suggest removing the `/movies` folder. The `cephfs` fs is only deleted and created when setting up class. Remove the residue can make life easier when adding/writing other tests in the same class.

Reply:
Done